packages/react-native-renderer/src/ReactNativeBridgeEventPlugin.js

```diff
@@ -76,14 +81,26 @@ export function traverseTwoPhase(inst: Object, fn: Function, arg: Function) {
   for (i = path.length; i-- > 0; ) {
     fn(path[i], 'captured', arg);
   }
-  for (i = 0; i < path.length; i++) {
-    fn(path[i], 'bubbled', arg);
+  if (skipBubbling) {
+    // Dispatch on target only
+    fn(path[0], 'bubbled', arg);
+  } else {
+    for (i = 0; i < path.length; i++) {
+      fn(path[i], 'bubbled', arg);
+    }
   }
 }
 
 function accumulateTwoPhaseDispatchesSingle(event) {
   if (event && event.dispatchConfig.phasedRegistrationNames) {
-    traverseTwoPhase(event._targetInst, accumulateDirectionalDispatches, event);
+    const skipBubbling = !!event.dispatchConfig.phasedRegistrationNames
```

Comment:
It might make the code easier to read by moving this branching logic to `extractEvents` on line 183. For example, maybe create a new `accumulateCapturePhaseDispatches(events)` method (but still reusing `traverseTwoPhase`).

Reply:
@lunaleaps my PR also changes this a little bit to centralize some duplicated code, do you want to rebase on top of that? Not sure how much it will impact your implementation here

Reply:
I don't think it's impacted, since you'll probably land first, happy to adjust on top of your changes once it's on main